M2/Macaulay2/m2/matrix1.m2

```diff
@@ -491,11 +491,11 @@ Ideal#{Standard,AfterPrint} = Ideal#{Standard,AfterNoPrint} = (I) -> (
 Ideal ^ ZZ := Ideal => (I,n) -> ideal symmetricPower(n,generators I)
 Ideal * Ideal := Ideal => ((I,J) -> ideal flatten (generators I ** generators J)) @@ samering
 Ideal * Module := Module => ((I,M) -> subquotient (generators I ** generators M, relations M)) @@ samering
-dim Ideal := I -> dim cokernel generators I
+dim Ideal := (cacheValue symbol dim) (I -> dim cokernel generators I)
```

Comment:
It's probably unnecessary to cache here and for `dim Module`. Since `cokernel` and `generators` are both cached, caching at `dim Module` should suffice.

The reason I say this is that `CacheFunctions` make debugging a bit complicated, so trying to use them only at the bottlenecks is better.

Reply:
You are right, this does not seem to improve much.